src/statement-list/statement_list_entry_point.cpp

```diff
@@ -79,6 +82,33 @@ static void generate_statement_list_init_function(symbol_tablet &symbol_table)
   symbol_table.add(init);
 }
 
+/// Creates __CPROVER_rounding_mode and adds it to the symbol table.
+/// Initializes the rounding mode via __CPROVER_initialize.
+/// \param [out] symbol_table: Symbol table that should contain the symbol and
+///   already contains __CPROVER_initialize.
+static void generate_rounding_mode(symbol_tablet &symbol_table)
+{
+  symbolt &init = symbol_table.get_writeable_ref(INITIALIZE_FUNCTION);
+  symbolt rounding_mode;
+  rounding_mode.name = ROUNDING_MODE_NAME;
+  rounding_mode.type = signed_int_type();
+  rounding_mode.is_thread_local = true;
+  rounding_mode.is_static_lifetime = true;
+  symbol_table.add(rounding_mode);
+
+  code_blockt &block = to_code_block(to_code(init.value));
+  const constant_exprt rounding_val{
+    std::to_string(ieee_floatt::rounding_modet::ROUND_TO_EVEN),
+    signed_int_type()};
+  code_assignt assign{rounding_mode.symbol_expr(), rounding_val};
+  block.add(assign);
+
+  code_blockt dest;
+  dest.add(code_labelt(CPROVER_HIDE, code_skipt()));
+  rounding_mode.value = dest;
```

Comment:
The value of `rounding_mode` should just be `rounding_val`, not a `code_blockt`.